docs/portal/developer-portal/advanced/Enabling_K3s.md

```diff
@@ -0,0 +1,317 @@
+
+### Use of Podman
+Traditional UAIs in CSM required some level of privilege in CSM for access to host volume mounts, networking, and startup activities. Podman offers an attractive solution for an interactive environment in which to place users, they can be rootless containers that do not rely on privilege escalation. When running on UANs, podman containers have access to a hosting environment that is already tailored to users.
+
+### Overview
+The overall component flow for replicating containerized environments for End-Users on UANs is as follows:
+- A user uses `ssh` to initiate a connection the HAProxy load-balancer running in K3s.
+- HAProxy, using the configured load balancing algorithms, will forward the SSH connection to an instance of `sshd`
+- `sshd`, running on a UAN via systemd, will initiate a rootless podman container as the user using the `ForceCommand` configuration.
+- The user is placed in a podman container for an interactive session, or their `SSH_ORIGINAL_COMMAND` is run in the container.
+- When the user disconnects, the podman process exits, and the container is removed.
+
+There are alternate configurations of podman that would allow for different workflows, for example, the main pid of the container could be long running, to facilitate easier reentry to the container on subsequent logins. 
+
+## Prerequisites 
+
+The following sections should be completed prior to configuring the UAN with K3s.
+
+### UAN K3s Node
+Designate a UAN to operate as the K3s control-plane node.
+
+**Note**: In a future release, additional UANs will be able to join as extra manager or worker nodes. 
+
+### MetalLB IP Address Pool
+
+A pool of IP Address should be configured for services running in K3s. This will allow for external `LoadBalancer` IP Address to be assigned to services like `HAProxy`.  This address pool must be routable from the UAN control-plane, and should be unused for other purposes. Initially, these IP address will serve as the SSH ingress for instances of `HAProxy`.
+
+### Subuid and Subgid
+
+To allow for users to run rootless podman containers, the files `/etc/subuid` and `/etc/subgid` must be present and configured with an entry for each user. These files should be uploaded to the `user` S3 bucket:
+```bash
+$ cray artifacts list user --format json
+{
+  "artifacts": [
+    {
+      "Key": "subuid",
+      "LastModified": "2023-02-21T23:41:43.948000+00:00",
+      "ETag": "\"c543aebb9b40bcf48879885734447090\"",
+      "Size": 145686,
+      "StorageClass": "STANDARD",
+      "Owner": {
+        "DisplayName": "User Service User",
+        "ID": "USER"
+      }
+    },
+    {
+      "Key": "subgid",
+      "LastModified": "2023-02-21T23:41:43.948000+00:00",
+      "ETag": "\"73032ede132e44d2c1bc567246901737\"",
+      "Size": 145686,
+      "StorageClass": "STANDARD",
+      "Owner": {
+        "DisplayName": "User Service User",
+        "ID": "USER"
+      }
+    }
+  ]
+}
+```
```

Comment:
```suggestion
The following steps should be completed prior to configuring the UAN with K3s.

1. Designate a UAN to operate as the K3s control-plane node.

   **Note**: In a future release, additional UANs will be able to join as extra manager or worker nodes. 

1. Identify a pool of IP addresses for the services running in K3s.
   
   This address pool must be routable from the UAN control-plane, and should be unused for other purposes.

   This will allow for external `LoadBalancer` IP Address to be assigned to services like `HAProxy`. Initially, these IP addresses will serve as the SSH ingress for instances of `HAProxy`.

1. Configure and create the `/etc/subuid` and `/etc/subgid` files.

   To allow for users to run rootless podman containers, these files must be present and configured with an entry for each user. These files should be uploaded to the `user` S3 bucket:

   ```bash
   $ cray artifacts list user --format json
   {
     "artifacts": [
       {
         "Key": "subuid",
         "LastModified": "2023-02-21T23:41:43.948000+00:00",
         "ETag": "\"c543aebb9b40bcf48879885734447090\"",
         "Size": 145686,
         "StorageClass": "STANDARD",
         "Owner": {
           "DisplayName": "User Service User",
           "ID": "USER"
        }
       },
       {
         "Key": "subgid",
         "LastModified": "2023-02-21T23:41:43.948000+00:00",
         "ETag": "\"73032ede132e44d2c1bc567246901737\"",
         "Size": 145686,
         "StorageClass": "STANDARD",
         "Owner": {
           "DisplayName": "User Service User",
           "ID": "USER"
         }
       }
     ]
   }
   ```
```